tests/fixtures/optimizer/qualify_columns.sql

```diff
@@ -0,0 +1,124 @@
+--------------------------------------
+-- Qualify columns
+--------------------------------------
+SELECT a FROM x;
+SELECT x.a AS a FROM x;
+
+SELECT a AS a FROM x;
+SELECT x.a AS a FROM x;
+
+SELECT x.a FROM x;
+SELECT x.a AS a FROM x;
+
+SELECT x.a AS a FROM x;
+SELECT x.a AS a FROM x;
+
+SELECT a AS b FROM x;
+SELECT x.a AS b FROM x;
+
+SELECT 1, 2 FROM x;
+SELECT 1 AS "_col0", 2 AS "_col1" FROM x;
+
+--------------------------------------
+-- Derived tables
+--------------------------------------
+SELECT y.a AS a FROM (SELECT x.a AS a FROM x) AS y(a);
+SELECT y.a AS a FROM (SELECT x.a AS a FROM x) AS y(a);
+
+SELECT y.a FROM (SELECT a AS a FROM x) AS y(a);
+SELECT y.a AS a FROM (SELECT x.a AS a FROM x) AS y(a);
+
+SELECT y.a FROM (SELECT x.a AS a FROM x) AS y;
+SELECT y.a AS a FROM (SELECT x.a AS a FROM x) AS y(a);
+
+SELECT a FROM (SELECT a AS a FROM x) y;
+SELECT y.a AS a FROM (SELECT x.a AS a FROM x) AS y(a);
+
+SELECT a FROM (SELECT a AS a FROM x);
+SELECT "_alias0".a AS a FROM (SELECT x.a AS a FROM x) AS "_alias0"(a);
```

Comment:
can we call it "_q_0" or "_query_0" ?

Reply:
for select x + 1, will you make the alias "_col_0" ?

Reply:
Yeah.

I'm not sure if there's a standard way to generate projection aliases? Some engines seem to have some rule for stringifying the selection... but an incrementing string felt easier.

Reply:
some alias as x + 1 AS "x + 1" but this isn't unique or good